src/gather.jl

```diff
@@ -0,0 +1,86 @@
+export gather, gather!
+
+"""
+    gather!(dst, src, idx, dims)
```

Comment:
Should `dims` be made a keyword arg, on user-facing functions?
```suggestion
    gather!(dst, src, idx; dims=0)
```

Reply:
Check new commits please.